datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -0,0 +1,171 @@
+from datasets import Mode
+from datasets.context import Context
+from datasets.dataset_plugin import DatasetPlugin
+
+
+if TYPE_CHECKING:
+    import dask.dataframe as dd
+    import pyspark
+
+
+class InvalidOperationException(Exception):
+    pass
+
+
+@DatasetPlugin.register_plugin(constructor_keys={"name"}, context=Context.Batch)
+class BatchDatasetPlugin(DatasetPlugin):
+    """
+    This is the default plugin for the Batch execution context.
+    """
+
+    _dataset_path_func: callable = None
+
+    def __init__(
+        self,
+        name: str,
+        logical_key: str = None,
+        columns=None,
+        run_id=None,
+        mode: Mode = Mode.Read,
+        path: Optional[str] = None,
+        partition_by: Optional[str] = None,
+        attribute_name: Optional[str] = None,
+    ):
+        self.path = path
+        self.partition_by = partition_by
+        self.program_name = self._executor.current_program_name
+        super(BatchDatasetPlugin, self).__init__(
+            name=name,
+            logical_key=logical_key,
+            columns=columns,
+            run_id=run_id,
+            mode=mode,
+            attribute_name=attribute_name,
+        )
+
+    def _get_path_filters_columns(self, columns, run_id: Optional[str] = None) -> Tuple[str, list, list[str]]:
+        path = self._get_dataset_path()
+        read_columns = self._get_read_columns(columns)
+        filters = None
+        query_run_id = run_id if run_id else self.run_id
+        if query_run_id:
+            filters = [("run_id", "=", query_run_id)]
+        return path, filters, read_columns
+
+    def read_pandas(
+        self,
+        columns: Optional[str] = None,
+        storage_format: str = "parquet",
+        run_id: Optional[str] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        path, filters, read_columns = self._get_path_filters_columns(columns, run_id=run_id)
+
+        df: pd.DataFrame
+        if storage_format == "parquet":
+            df = pandas.read_parquet(path, columns=read_columns, engine="pyarrow", filters=filters, **kwargs)
+        elif storage_format == "csv":
+            df = pandas.read_csv(path, columns=read_columns, filters=filters, **kwargs)
+
+        for meta_column in self._META_COLUMNS:
+            if meta_column in df and (read_columns is None or meta_column not in read_columns):
+                del df[meta_column]
+        return df
+
+    def write(self, data: pd.DataFrame, **kwargs):
+        if not (self.mode & Mode.Write):
+            raise InvalidOperationException(f"Cannot write because mode={self.mode}")
+
+        if not isinstance(data, pd.DataFrame):
+            assert ValueError("data is not a pandas DataFrame")
+
+        if self.partition_by:
+            partition_cols = self.partition_by.split(",")
+        else:
+            partition_cols = list()
+
+        if self.path is None or "run_id" in partition_cols:
+            # Only partition on run_id if @dataset(path="s3://..") is given
```

Comment:
is this comment right? should it be "Only partition on run_id if @dataset(path="s3://..") is not given?

Reply:
I fixed the comment in this PR https://github.com/zillow/datasets/pull/6/files
thanks!